### PR TITLE
block tag latest for images

### PR DESCRIPTION
**Pull request by igorkchyts**
Disable latest tag for all namespaces except openshift-marketplace. - I am not sure why image is latest in this project but guess that it should be always looking for latest version in order to keep operators up to date.

### CODE REVIEW

File: open-policy-agent/trusted-image-sources/disallowedtags/template.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+apiVersion: templates.gatekeeper.sh/v1beta1
+kind: ConstraintTemplate
+metadata:
+  name: k8sdisallowedtags
+  annotations:
+    description: Requires container images to have an image tag different
+      from the ones in a specified list.
```
Comment (cloudlabmaz) [inline open-policy-agent/trusted-image-sources/disallowedtags/template.yaml:7]: and prevent empty tag

File: open-policy-agent/trusted-image-sources/disallowedtags/template.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+          tag := [contains(container.image, ":")]
+          not all(tag)
+          msg := sprintf("container <%v> didn't specify an image tag <%v>", [container.name, container.image])
+        }
+        input_containers[c] {
+          c := input.review.object.spec.containers[_]
+        }
+        input_containers[c] {
+          c := input.review.object.spec.initContainers[_]
+        }
```
Comment (cloudlabmaz) [inline open-policy-agent/trusted-image-sources/disallowedtags/template.yaml:42]: Very Cool

File: open-policy-agent/trusted-image-sources/disallowedtags/constraint.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+kind: K8sDisallowedTags
+metadata:
+  name: container-image-must-not-have-latest-tag
+spec:
+  match:
+    kinds:
+      - apiGroups: [""]
+        kinds: ["Pod"]
+    excludedNamespaces:
+      - "openshift-marketplace"
```
[the remark this answers is not in the document]
Comment (igorkchyts) [reply open-policy-agent/trusted-image-sources/disallowedtags/constraint.yaml:11]: Yes. The community-operator using the tag 'latest' by default. There is a sense to use that in connected environments, but also may be disabled in the disconnected.

File: open-policy-agent/trusted-image-sources/disallowedtags/template.yaml (hunk 2 above)
Comment (igorkchyts) [reply open-policy-agent/trusted-image-sources/disallowedtags/template.yaml:42]: Thanks @cloudlabmaz

**Review by michaelkotelnikov — APPROVED** [review]

**Pull request merged**